wifiutils/src/main/java/com/thanosfisherman/wifiutils/WifiUtils.java

```diff
@@ -147,9 +153,10 @@ public void onScanResultsReady() {
                 }
             }
             if (mSingleScanResult != null && mPassword != null) {
-                if (connectToWifi(mContext, mWifiManager, mSingleScanResult, mPassword)) {
-                    registerReceiver(mContext, mWifiConnectionReceiver.activateTimeoutHandler(mSingleScanResult),
-                            new IntentFilter(WifiManager.SUPPLICANT_STATE_CHANGED_ACTION));
+                if (connectToWifi(mContext, mWifiManager, mConnectivityManager, mSingleScanResult, mPassword, mWifiConnectionCallback)) {
+                    // FIXME: get timeoutHandler out of this?
+                    registerReceiver(mContext, ((ConnectionHandler)mWifiConnectionReceiver).connectWith(mSingleScanResult, mPassword, mConnectivityManager),
```

Comment:
Thanks for the feedback. At the moment we have duplicate timeout logic because it's in the broadcast receivers. Because there are 2 now, it would better be somewhere apart. I'll have a try.